getrepodata.py

```diff
@@ -0,0 +1,88 @@
+import asyncio
+from collections import defaultdict
+
+import tqdm
+import pydriller
+import httpx
+import itertools
+
+headers = {'Authorization': 'token '}
+
+
+async def fetch_stargazer(client: httpx.AsyncClient, url, page):
+    response = await client.get(url.format(page), headers=headers)
+    # print(response.json())
+    return response.json()
+
+
+async def get_stargazers_for_page(url):
+    stargazers = set()
+    async with httpx.AsyncClient() as client:
+        response = await asyncio.gather(
+            *map(fetch_stargazer, itertools.repeat(client), itertools.repeat(url), list(range(1, stargazer_pages_num)))
+        )
+        for user in response:
+            stargazers.update(list(map(lambda user: user["login"], user)))
```

Comment:
set can be updated from iterable like generator
```suggestion
            stargazers.update(map(lambda user: user["login"], user))
```